src/OpenLoco/src/Vehicles/VehicleManager.cpp

```diff
@@ -56,12 +57,63 @@ namespace OpenLoco::VehicleManager
         }
     }
 
+    // 0x004C39D4
+    static void determineAvailableVehicleTypes(Company& company)
+    {
+        uint16_t availableTypes = 0;
+
+        for (uint32_t i = 0; i < ObjectManager::getMaxObjects(ObjectType::vehicle); ++i)
+        {
+            if (!(company.unlockedVehicles[i & 0x1F] & 1ULL << (i / 32)))
```

Comment:
It's probably better to create a new variable for the mask, the member is also 32 bit and not 64.